src/test/java/com/google/javascript/clutz/ctor_func.d.ts

```diff
@@ -1,5 +1,7 @@
 declare namespace ಠ_ಠ.clutz.ctor_func {
-  class Ctor < T > {
+  class Ctor < T > extends Ctor_Instance < T > {
```

Comment:
agree.
It will be a bit tricker, because now if Foo superclasses Ctor, it can't blindly expect _Instance to be there (i.e. we need some global tracking of what _Instance classes exist).

There is another venue of potential optimization - if there is no superclassing of Foo (even with statics) we dont need Foo_Instance.

#196